server/models/Application.ts

```diff
@@ -1,13 +1,60 @@
-import { randomBytes } from 'crypto';
+import type { CreationOptional, InferAttributes, InferCreationAttributes, ForeignKey, NonAttribute } from 'sequelize';
+
+import { crypto } from '../lib/encryption';
+import sequelize, { DataTypes, Model } from '../lib/sequelize';
 
+import User from './User';
+
+import { randomBytes } from 'crypto';
 import { isNil, merge } from 'lodash';
-import { DataTypes } from 'sequelize';
 
-import { crypto } from '../lib/encryption';
-import sequelize from '../lib/sequelize';
+export type ApplicationType = 'apiKey' | 'oAuth';
+
+class Application extends Model<InferAttributes<Application>, InferCreationAttributes<Application>> {
+  public declare readonly id: CreationOptional<number>;
+  public declare CollectiveId: number;
+  public declare CreatedByUserId: ForeignKey<User['id']>;
+  public declare type: ApplicationType;
+  public declare apiKey: string;
+  public declare clientId: string;
+  public declare clientSecret: string;
+  public declare callbackUrl:string;
+  public declare name: string;
+  public declare description: string;
+  public declare disabled: boolean;
+  public declare createdAt: CreationOptional<Date>;
+  public declare updatedAt: CreationOptional<Date>;
+  public declare deletedAt: CreationOptional<Date>;
+
+  protected get info(): NonAttribute<Partial<Application>> {
```

Comment:
Even though it can return sensitive data, this method is still meant to be called from the outside. It was public before, let's make sure we preserve features parity

```suggestion
  get info(): NonAttribute<Partial<Application>> {
```